### ISSUE TITLE
[AQP][L3 - On demmand] Develop L3 commands to read Air Quality sensor in demmand

### ISSUE THREAD

**Issue by ralcaide** [Sinapse-Energia/AP-Sinapse#160]
To develop the MQTT command to read Air Quality sensor(s) on demmand:

[L3 AQP On demmand](https://github.com/Sinapse-Energia/AP-Sinapse/wiki/%5BExternal-Sensors%5D-MQTT-Sinapse-API#air-quality--particles)

Based on [Generic UART](https://github.com/Sinapse-Energia/AP-Sinapse/issues/107)

**Comment by ralcaide** [comment]
@fpgamcu , could you give us an estimation about this task? Is everything clear?

**Comment by fpgamcu** [comment]
5 hours should be enough. Do you have datasheet on the sensors that would be used?

**Comment by ralcaide** [comment]
@fpgamcu, this command have a high priority right now. Please focus on this one at the beginning of the current sprint.

When could we have a first testable version?

**Comment by fpgamcu** [comment]
Sure, now working on it. Thursday evening I will push the code.

**Comment by ralcaide** [comment]
Ok, so tomorrow evening, we will plan first test on Friday. Thank you

**Comment by ralcaide** [comment]
Great, we will provide you with an external with the sensor

**Comment by fpgamcu** [comment]
To be 100% sure, On Demand is performed like next:
1) Triggering the measurement
2) Wait 6s for result
3) Send result over MQTT

Is it ok, or the operation should be performed in different way?

**Comment by ralcaide** [comment]
Yes, but the values should be saved in NVM with timestamp. Next time, if the sensor does not answer should send last measurement

**Comment by fpgamcu** [comment]
Ok, several more questions:
1) If we specify for example M6, should other parameters remain as was, or changed to X? For example what is true:
a) L3_OD_AQP;M6; -> L3_OD_AQP_R;123456;X;X;X;X;X;3;X;1234;
b) L3_OD_AQP;M6; -> L3_OD_AQP_R;123456;X;X;X;X;1;3;2;1234;
2) If we specify ALL, should timestamp be changed if any parameters was updated, or if all was updated:
what is true:
a) L3_OD_AQP;ALL; -> (TS was 1234, answered only M2) -> L3_OD_AQP_R;123456;X;2;X;X;1;3;4;1235;
b) L3_OD_AQP;ALL; -> (TS was 1234, answered only M2) -> L3_OD_AQP_R;123456;X;2;X;X;1;3;4;1234;

**Comment by ralcaide** [comment]
1:a), 2: a)

**Comment by fpgamcu** [comment]
ok, so in L3_OD_AQP_R; not needed parameters should be changed to X in response, but remain as was in NVM.

**Comment by fpgamcu** [comment]
also now there is only one timestamp for all parameters, so we could have situation like next:
0) TS was 1230
1) L3_OD_AQP;M6;
2) got correct answer over uart
3) L3_OD_AQP_R;123456;X;X;X;X;X;3;X;1234;
4) L3_OD_AQP;M7;
2) got incorrect answer over uart or no answer
3) L3_OD_AQP_R;123456;X;X;X;X;X;X;X;1234;
is it ok?

**Comment by ralcaide** [comment]
It's strange but we leave like that by the moment, let me think

**Comment by ralcaide** [comment]
Tested OK in V1.30.2
